source/exercises/mailroom-oo.rst

```diff
@@ -51,29 +52,64 @@ One of the hardest parts of OO design (particularly in Python) is to know how "l
 
 There are no hard and fast rules, but here are some guidelines:
 
-For this assignment it's OK to go with simple tuples. However, in order for the code to be more flexible in the future, for new "fields" were added to the donor object, it's probably better to use a more structured type, so you don't have to worry about changing the order or number of fields.
+For this assignment it's OK to go with simple tuples. However, in order for the code to be more flexible in the future, if new "fields" were added to the donor object, it's probably better to use a more structured type, so you don't have to worry about changing the order or number of fields.
 
 So now you have to think about using a dict or class? Again for flexibility, I think a dict is a bit easier -- you can add fields to it very easily. However, with a class, you can build some functionality in there, too. This is where encapsulation comes in -- for instance, one thing you might want to do is get the total of all donations a donor has made in the past. If you add a method to compute that (or a property!), then the rest of the code doesn't need to know how the donations are stored.
 
-So: if you want my hints :-) ...
+Consider ``data[0]`` vs ``data["first_name"]`` vs ``data.first_name`` -- which one is more readable? Keep in mind that another benefit of using OO for data encapsulation is ability of modern IDE to provide auto-completion, which reduces number of bugs and helps to produce code faster.
+
+Below are more detailed suggestions on breaking down your existing code into multiple modules that will be part of a single mailroom program.
+
+Donor Class
+^^^^^^^^^^^^
+
+Module responsible for donor data encapsulation.
 
-You'll want a Donor class -- this will hold all the information about the donor, and have attributes, properties, and methods to provide access to the donor specific information that is needed.
+This will hold all the information about a single donor, and have attributes, properties, and methods to provide access to the donor specific information that is needed.
+Remember that if you are writing code that only accesses information about a single donor then it should most likely live in this class.
 
-You'll then want a class that handles the collection of donors. This will hold all the donor objects, but also methods to add a new donor, search for a given donor, etc. If you want a way to save and re-load your data, this class would have that too.
+DonorCollection Class
+^^^^^^^^^^^^^^^^^^^^^
+
+Module responsible for donor collection data encapsulation.
```

Comment:
we should be clear about what is in a "module" vs a "class".

In python, you can have one class per module, but that's not necessarily a common practice.

But in any case, we should be clear in the description.

Reply:
Do you think this should be renamed to Class instead of module? Or explanation at the top about module vs class?